libs/actions/sendmail.c

```diff
@@ -0,0 +1,226 @@
+	struct JsonNode *jvalues2 = NULL;
+	struct JsonNode *jvalues3 = NULL;
+	struct JsonNode *jval1 = NULL;
+	struct JsonNode *jval2 = NULL;
+	struct JsonNode *jval3 = NULL;
+	struct mail_t mail;
+	char *message = NULL, *to = NULL, *subject = NULL;
+	char *sfrom = NULL, *sto = NULL, *ssubject = NULL;
+	char *shost = NULL, *suser = NULL, *spassword = NULL;
+	int sport= 0;	
+
+	// mail settings
+	settings_find_string("mail-from", &sfrom);
+	settings_find_string("mail-to", &sto);
+	settings_find_string("mail-subject", &ssubject);
+
+	//smtp settings
+	settings_find_string("smtp-host", &shost);
+	settings_find_number("smtp-port", &sport);
+	settings_find_string("smtp-user", &suser);
+	settings_find_string("smtp-password", &spassword);
+
+	// arguments
+	jsubject = json_find_member(arguments, "SUBJECT");
+	jmessage = json_find_member(arguments, "MESSAGE");
+	jto = json_find_member(arguments, "TO");
+	
+	if(jmessage != NULL) {	
+		if((jvalues1 = json_find_member(jmessage, "value")) != NULL) {
+			if((jval1 = json_find_element(jvalues1, 0)) != NULL && jval1->tag == JSON_STRING){
+				message = jval1->string_;
+			}
+			subject = ssubject;
+			if(jsubject != NULL) {	
+				if((jvalues2 = json_find_member(jsubject, "value")) != NULL) {
+					if((jval2 = json_find_element(jvalues2, 0)) != NULL && jval2->tag == JSON_STRING){
+						subject = jval2->string_;
+					} 
+				}
+			}
+			to = sto;
+			if(jto != NULL) {	
+				if((jvalues3 = json_find_member(jto, "value")) != NULL) {
+					if((jval3 = json_find_element(jvalues3, 0)) != NULL && jval3->tag == JSON_STRING){
+						to = jval3->string_;
+					}
+				}
+			}
+			
+			mail.from = MALLOC(strlen(sfrom)+1);
+			mail.to = MALLOC(strlen(to)+1);
+			mail.subject = MALLOC(strlen(subject)+1);
+			mail.message = MALLOC(strlen(message)+1);
+
+			strcpy(mail.from, sfrom);
+			strcpy(mail.to, to);
+			strcpy(mail.subject, subject);
+			strcpy(mail.message, message);
+			
+			if(sendmail(shost, suser, spassword, sport, &mail) != 0) {
+				logprintf(LOG_DEBUG, "Sendmail failed with message \"%s\"", message);
+			}
+			
+			FREE(mail.from);
+			FREE(mail.to);
+			FREE(mail.subject);
+			FREE(mail.message);
+		}
+	}
+	return 0;
+}
+
+#ifndef MODULE
+__attribute__((weak))
+#endif
+void actionSendmailInit(void) {
+	event_action_register(&action_sendmail, "sendmail");
+
+	options_add(&action_sendmail->options, 'a', "SUBJECT", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_STRING, NULL, NULL);
+	options_add(&action_sendmail->options, 'b', "MESSAGE", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
+	options_add(&action_sendmail->options, 'c', "TO", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_STRING, NULL, NULL);
+
+	action_sendmail->run = &actionSendmailRun;
+	action_sendmail->checkArguments = &actionSendmailArguments;
+}
+
+#ifdef MODULE
+void compatibility(struct module_t *module) {
+	module->name = "sendmail";
+	module->version = "1.0";
+	module->reqversion = "6.0";
+	module->reqcommit = NULL;
```

Comment:
I know you introduced the mail library in 6e2e41d and tried to get the 
commit number the way you indicated, but I get this:

root@raspi2:/home/pi/pilight# git checkout 6e2e41d
HEAD is now at 6e2e41d... Various fixes
root@raspi2:/home/pi/pilight# git describe
fatal: No tags can describe '6e2e41dd7bc241fc48e1f4a638da173d71789cc3'.

I suppose the reqcommit must be set to a number that should be 
represented by a tag?

CurlyMoo schreef op 17-3-2015 om 19:07:
>
> In libs/actions/sendmail.c 
> <https://github.com/pilight/pilight/pull/155#discussion_r26599294>:
>
> > +	event_action_register(&action_sendmail, "sendmail");
> > +
> > +	options_add(&action_sendmail->options, 'a', "SUBJECT", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_STRING, NULL, NULL);
> > +	options_add(&action_sendmail->options, 'b', "MESSAGE", OPTION_HAS_VALUE, DEVICES_VALUE, JSON_STRING, NULL, NULL);
> > +	options_add(&action_sendmail->options, 'c', "TO", OPTION_HAS_VALUE, DEVICES_SETTING, JSON_STRING, NULL, NULL);
> > +
> > +	action_sendmail->run = &actionSendmailRun;
> > +	action_sendmail->checkArguments = &actionSendmailArguments;
> > +}
> > +
> > +#ifdef MODULE
> > +void compatibility(struct module_t *module) {
> > +	module->name = "sendmail";
> > +	module->version = "1.0";
> > +	module->reqversion = "6.0";
> > +	module->reqcommit = NULL;
>
> Please check which commit included the sendmail library.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/pilight/pilight/pull/155/files#r26599294>.
>